### PR TITLE
feat: Add more filter functions

### PR DESCRIPTION
**Pull request by marikaner**
## Context

Implemented all filter functions that are currently possible

## Definition of Done

Please consider all items and remove only if not applicable.

- [ ] Tests created/adjusted for your changes.
- [ ] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [ ] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [ ] If applicable: Properly documented (JSDoc of public API)
- [ ] If applicable: Check if `node run doc` still works.

### CODE REVIEW

File: packages/core/src/odata/common/filter/filter-function-base.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import { EdmTypeShared } from '../edm-types';
+import { EntityBase, ODataVersionOf } from '../entity';
+import { Field, FieldType } from '../selectable';
+import { Filter } from './filter';
+
+/**
+ * Data structure to represent OData V2 filter functions.
```
Comment (FrankEssenberger) [inline packages/core/src/odata/common/filter/filter-function-base.ts:9]: Is this true? If so this class and the related ones should not be in the common folder but in the V2 specific one.

File: packages/core/src/odata/common/filter/filter-function.ts [hunk 2]
```diff
@@ -1,97 +1,62 @@
...
-   * Serializes the filter function into a string
-   * @param parentFieldNames - Names of parents in case the function is part of a filter on a navigation property
-   * @returns The filter function as string
-   */
-  toString(parentFieldNames: string[] = []): string {
-    const params = this.parameters
-      .map(param => this.transformParameter(param, parentFieldNames))
-      .join(', ');
-    return `${this.functionName}(${params})`;
+export function filterFunction<EntityT extends EntityBase>(
```
Comment (FrankEssenberger) [inline packages/core/src/odata/common/filter/filter-function.ts:35]: I would call this method a bit different to indicate, that it builds, construcuts filterfunction but is not itself a filter function.

File: test-packages/integration-tests/test/v2/filter-functions.spec.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import {
```
Comment (FrankEssenberger) [inline test-packages/integration-tests/test/v2/filter-functions.spec.ts:2]: genral comment: In the `filter-factory.ts` of the test util I see many filter object beeing created for testing. But this has not been adjusted for all the new filter function beeing defined here. Is this obvious?

Comment (FrankEssenberger) [reply test-packages/integration-tests/test/v2/filter-functions.spec.ts:2]: As discussed: Only do the specific cases.
Add type tests to `filter.spec.ts`.

File: packages/core/src/odata/common/filter/filter-function-base.ts (hunk 1 above)
Comment (marikaner) [reply packages/core/src/odata/common/filter/filter-function-base.ts:9]: No it's not. It is a left over from before.

File: packages/core/src/odata/common/filter/filter-function.ts (hunk 2 above)
Comment (marikaner) [reply packages/core/src/odata/common/filter/filter-function.ts:35]: Phew, this function has been there for a while now. I think as it is supposed to also be used by our users the name is quite good. I would generally like to keep it, but independent of that this is out of scope for this PR is guess ;)

File: test-packages/integration-tests/test/v2/filter-functions.spec.ts (hunk 3 above)
Comment (marikaner) [reply test-packages/integration-tests/test/v2/filter-functions.spec.ts:2]: Added some more tests as discussed.

**Pull request merged**